torchmetrics/metric.py

```diff
@@ -221,6 +231,72 @@ def wrapped_func(*args, **kwargs):
 
         return wrapped_func
 
+    def sync(
+        self,
+        dist_sync_fn: Optional[Callable] = None,
+        process_group: Optional[Any] = None,
+        should_sync: bool = True,
+    ) -> Dict[str, Tensor]:
+        """
+        Sync function for manually controlling when metrics states should be synced across processes
+
+        Args:
+            dist_sync_fn: Function to be used to perform states synchronization
+            process_group:
+                Specify the process group on which synchronization is called.
+                default: None (which selects the entire world)
+            should_sync: Whether to apply to state synchronization.
+
+        Returns:
+            cache: A dictionarry containing the local metric states. The cache will be empty if sync didn't happen.
+        """
+        is_distributed = torch.distributed.is_available() and torch.distributed.is_initialized()
+
+        if dist_sync_fn is None:
+            # User provided a bool, so we assume DDP if available
+            dist_sync_fn = gather_all_tensors
+
+        cache = {}
+
+        if is_distributed and should_sync and dist_sync_fn is not None:
```

Comment:
We know it's not `None` at this point as it would have been overridden a few lines before

```suggestion
        if is_distributed and should_sync:
```

Reply:
Should we early exit if `not should_sync`?